### DISCUSSION TITLE
retry_delay issue !!

### DISCUSSION THREAD

**Discussion by wezo3** [savonet/liquidsoap#2716] [General]
Hello,
I have issue with retry_delay its not working with my code example 

```
def next() =
        uri = list.hd(process.read.lines("php myscript.php"))
        request.create(uri)
end

radio= request.dynamic(id="input",retry_delay=5.,next)

```
when audio file not load my script print no-data

and liquidsoap still retry 4  or 5 time in same second and this effect to load on the server
and I need to retry one time every 5 seconds
like this in the log :

```
2022/10/31 06:48:22 [request:3] Nonexistent file or ill-formed URI "no-data"!
2022/10/31 06:48:23 [request:3] Nonexistent file or ill-formed URI "no-data"!
2022/10/31 06:48:23 [request:3] Nonexistent file or ill-formed URI "no-data"!
2022/10/31 06:48:23 [request:3] Nonexistent file or ill-formed URI "no-data"!
2022/10/31 06:48:23 [request:3] Nonexistent file or ill-formed URI "no-data"!
2022/10/31 06:48:23 [request:3] Nonexistent file or ill-formed URI "no-data"!
2022/10/31 06:48:24 [request:3] Nonexistent file or ill-formed URI "no-data"!
2022/10/31 06:48:24 [request:3] Nonexistent file or ill-formed URI "no-data"!
2022/10/31 06:48:24 [request:3] Nonexistent file or ill-formed URI "no-data"!
2022/10/31 06:48:24 [request:3] Nonexistent file or ill-formed URI "no-data"!
```

how fix this ?

**Comment by wezo3** [comment]
@toots 
can you please help and let me know this is bug or wrong in my example code ?

**Reply by vitoyucepi** [reply]
Hi @wezo3,
Could you please post examples of `php myscript.php` output for success and failure?
I think in case of failure, your script will return some string instead of `null`.

You can check this using this script.
```ruby
def next() =
        print(time.local())
        null()
end

output.dummy(fallible=true, request.dynamic(id="input",retry_delay=5.,next))
```
It will print `time` every 5 seconds.

I recommend you output a JSON from your PHP script and then parse it with liquidsoap.

<details>
<summary>Success</summary>

```json
{
	"status": true,
	"file": "/path/to/file"
}
```

</details>

<details>
<summary>Failure</summary>

```json
{
	"status": false,
	"file": null
}
```

</details>

Or maybe utilize something like https://github.com/omniti-labs/jsend.

**Reply by wezo3** [reply]
thank you for response 
this is results for my php script
when there media to play
php myscript.php
/path/files/media.mp3

and when not any media will return this
no-data

i will try as you send about Success and Failure example and let you know after test
Thank you

**Reply by vitoyucepi** [reply]
Check page 77 of [the book](http://www.liquidsoap.info/book/book.pdf).

**Reply by wezo3** [reply]
thank you very much, its works when I did like the page 77 in the book
Thanks again for your help
